zerver/models.py

```diff
@@ -1682,12 +1681,11 @@ def to_dict(self):
     @staticmethod
     def status_from_string(status):
         # type: (NonBinaryStr) -> Optional[int]
+        status_val = None
         if status == 'active':
             status_val = UserPresence.ACTIVE
         elif status == 'idle':
             status_val = UserPresence.IDLE
-        else:
-            status_val = None
```

Comment:
I believe I've seen discussion of having this situation resolve to a Union[x, y, ...] based on the result for each branch of the conditional, but my understanding is this is not fixed in mypy, though possibly related to python/mypy#2611. In the revised commit, I've marked the first branch as `Optional[int]` with link to that issue.